src/color.rs

```diff
@@ -114,28 +187,275 @@ impl ToCss for RGBA {
-            dest.write_str(", ")?;
 
-            // Try first with two decimal places, then with three.
-            let mut rounded_alpha = (self.alpha_f32() * 100.).round() / 100.;
-            if clamp_unit_f32(rounded_alpha) != self.alpha {
-                rounded_alpha = (self.alpha_f32() * 1000.).round() / 1000.;
+        self.serialize_alpha(dest)?;
+
+        dest.write_char(')')
+    }
+}
+
+// NOTE: LAB and OKLAB is not declared inside the macro, because it causes
+// cbindgen to ignore them.
+
+/// Color specified by lightness, a- and b-axis components.
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[repr(C)]
+pub struct LAB {
+    /// The lightness component.
+    pub lightness: f32,
+    /// The a-axis component.
+    pub a: f32,
+    /// The b-axis component.
+    pub b: f32,
+    /// The alpha component.
+    pub alpha: u8,
+}
+
+/// Color specified by lightness, a- and b-axis components.
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[repr(C)]
+pub struct OKLAB {
+    /// The lightness component.
+    pub lightness: f32,
+    /// The a-axis component.
+    pub a: f32,
+    /// The b-axis component.
+    pub b: f32,
+    /// The alpha component.
+    pub alpha: u8,
+}
+
+macro_rules! impl_lab_like {
+    ($cls:ident, $fname:literal) => {
+        impl WithAlpha for $cls {
+            #[inline]
+            fn alpha(&self) -> u8 {
+                self.alpha
             }
+        }
 
-            rounded_alpha.to_css(dest)?;
+        impl $cls {
+            /// Construct a new Lab color format with lightness, a, b and alpha components.
+            pub fn new(lightness: f32, a: f32, b: f32, alpha: u8) -> Self {
+                Self {
+                    lightness,
+                    a,
+                    b,
+                    alpha,
+                }
+            }
+        }
+
+        #[cfg(feature = "serde")]
+        impl Serialize for $cls {
+            fn serialize<S>(&self, serializer: S) -> Result<serde::ser::Ok, serde::ser::Error>
+            where
+                S: Serializer,
+            {
+                (self.lightness, self.a, self.b, self.alpha).serialize(serializer)
+            }
+        }
+
+        #[cfg(feature = "serde")]
+        impl<'de> Deserialize<'de> for $cls {
+            fn deserialize<D>(deserializer: D) -> Result<Self, serde::de::Error>
+            where
+                D: Deserializer<'de>,
+            {
+                let (lightness, a, b, alpha) = Deserialize::deserialize(deserializer)?;
+                Ok(LAB::new(lightness, a, b, alpha))
+            }
+        }
+
+        impl ToCss for $cls {
+            fn to_css<W>(&self, dest: &mut W) -> fmt::Result
+            where
+                W: fmt::Write,
+            {
+                dest.write_str($fname)?;
+                dest.write_str("(")?;
+                self.lightness.to_css(dest)?;
+                dest.write_str(" ")?;
+                self.a.to_css(dest)?;
+                dest.write_str(" ")?;
```

Comment:
nit: `write_char(' ')` might be marginally more efficient, bunch of places in this patch.